Frameworks/CoreGraphics/CGColor.mm

```diff
@@ -31,25 +31,31 @@
  @Notes Limited constants supported
 */
 CGColorRef CGColorGetConstantColor(CFStringRef name) {
+    CGColorRef ret = nullptr;
     if (CFEqual(kCGColorBlack, name)) {
-        return [[__LazyUIColor blackColor] CGColor];
+        ret = static_cast<CGColorRef>([[__LazyUIColor blackColor] CGColor]);
     } else if (CFEqual(kCGColorWhite, name)) {
-        return [[__LazyUIColor whiteColor] CGColor];
+        ret = static_cast<CGColorRef>([[__LazyUIColor whiteColor] CGColor]);
     } else if (CFEqual(kCGColorClear, name)) {
-        return [[__LazyUIColor clearColor] CGColor];
+        ret = static_cast<CGColorRef>([[__LazyUIColor clearColor] CGColor]);
+    } else {
+        UNIMPLEMENTED_WITH_MSG("CGColorGetConstantColor does not support color %s", CFStringGetCStringPtr(name, kCFStringEncodingUTF8));
     }
 
-    UNIMPLEMENTED_WITH_MSG("CGColorGetConstantColor does not support color %s", CFStringGetCStringPtr(name, kCFStringEncodingUTF8));
-    return nullptr;
+    if (ret) {
+        auto grayColorSpace = woc::MakeStrongCF<CGColorSpaceRef>(CGColorSpaceCreateDeviceGray());
+        [static_cast<UIColor*>(ret) setColorSpace:grayColorSpace];
```

Comment:
the Caveat indicates it's limited. Should probably move it up so the 'future' is not clouded.